nozzle/preloaders.go

```diff
@@ -0,0 +1,88 @@
+package nozzle
+
+import (
+	"encoding/json"
+	"github.com/cloudfoundry-community/go-cfclient"
+	"io/ioutil"
+	"net/http"
+)
+
+// CFPreloader is a Preloader that reads directly from the CF API
+type CFPreloader struct {
+	client *cfclient.Client
+}
+
+// ExternalPreloader loads application info from an external source. An example is available here:
+// https://github.com/influxdata/influxdb-firehose-nozzle/tree/master/app-api-example
+type ExternalPreloader struct {
+	url string
+}
+
+// NewCFPreloader creates a new Preloader that wraps the supplied CF client
+func NewCFPreloader(client *cfclient.Client) *CFPreloader {
+	return &CFPreloader{
+		client: client,
+	}
+}
+
+// GetAllApps loads the entire list of applications
+func (c *CFPreloader) GetAllApps() ([]AppInfo, error) {
+	apps, err := c.client.ListApps()
+	if err != nil {
+		return nil, err
+	}
+	ai := make([]AppInfo, len(apps))
+	for i, a := range apps {
+		ai[i].Guid = a.Guid
+		ai[i].Name = a.Name
+		ai[i].Org = a.SpaceData.Entity.OrgData.Entity.Name
```

Comment:
Good to check for nil values before sourcing these. I believe we can do `a.Space()` and `space.Org` and it returns an error handle if these don't exist.